lib/papertrail/search_query.rb

```diff
@@ -2,20 +2,38 @@
 
 module Papertrail
   class SearchQuery
-    def initialize(connection, query = nil, options = {})
-      @connection = connection
-      @query      = query
-      @options    = options
+    def self.api_url
+      '/api/v1/events/search.json'
     end
 
-    def search
-      params = @options.dup
-      params[:q] = @query  if @query
-      params[:min_id] = @max_id if @max_id
+    def self.initial_search_limit
+      100
+    end
+
+    def self.subsequent_search_limits
+      1000
+    end
+
+    attr_reader :max_id, :not_first_request
 
-      response = @connection.get('/api/v1/events/search.json', params)
+    def search_results(connection, query = nil, options = {})
```

Comment:
Since `SearchQuery` is only created by a `Connection` and `connection` never changes between invocations, should it be moved back to a constructor arg?